### ISSUE TITLE
[Linux] Also supply AppImage?

### ISSUE THREAD

**Issue by guruz** [owncloud/client#5951]
```
<dragotin> https://build.opensuse.org/package/view_file/home:ocfreitag/owncloud-client/appimage.yml?expand=1
<dragotin> das ist die steuerdatei, die das appimage erzeugt. Es benutzt das owncloud-client paket und packt das um in ein appimage
```

**Comment by dragotin** [comment]
Something to check out: 
http://download.opensuse.org/repositories/home:/ocfreitag/AppImage/owncloud-client-latest-x86_64.AppImage

I hope to be able to fix the overlay icon issue for the gnome based browsers at least, with that it should be rather useful.

**Comment by michaelstingl** [comment]
👍  I'm open for an AppImage-Hack-Session at the [conference](https://occon17.owncloud.org/).

**Comment by ogoffart** [comment]
My current work in progress: https://hydro.woboq.im/owncloud/index.php/s/cQYOWcSw2Kdfw8Q
(this is based on https://github.com/OldFrostDragon/qt-5.7-xenial-docker , adapted for qt 5.9)

Also what nextcloud does:  https://github.com/nextcloud/client_theming/tree/master/linux/AppImage

Note that we'd need to find a solution for the shell integration. I think @dragotin mentioned that he has a solution for nautilus in his appimage. but i forgot what it was.  Was it to copy the python file to some user directory when starting the client?
For dolphin plugin, that might not be easy to solve.  We would still require distribution packages for these. Older version of the plugins should still work with newer version of the owncloud client so that might be good enough.

**Comment by michaelstingl** [comment]
I really like @guruz 's idea to ship the AppImage in a Linux repo and include all the things that aren't possible in the AppImage yet to work around the existing limitations.

**Comment by guruz** [comment]
> ship the AppImage in a Linux repo

Plus that gives you easy upgrading using apt-get or whatever:)

But I know that @ogoffart strongly disagrees.

**Comment by guruz** [comment]
Linking https://central.owncloud.org/t/appimage-for-desktop-client-test-it-now/7885/3

**Comment by koroki** [comment]
Dear. I'm using a computer where I cannot install nothing outside my home. The machine have Centos7. When I try to use the appImage, I obtain the following error:

Got keys from plugin meta data ("xcb")
loaded library "/tmp/.mount_ua6Xly/usr/lib64/qt5/plugins/platforms/libqxcb.so"
QLibraryPrivate::loadPlugin failed on "/tmp/.mount_ua6Xly/usr/lib64/qt5/plugins/platforms/libqxcb.so" : "Cannot load library /tmp/.mount_ua6Xly/usr/lib64/qt5/plugins/platforms/libqxcb.so: (libxkbcommon-x11.so.0: cannot open shared object file: No such file or directory)"
This application failed to start because it could not find or load the Qt platform plugin "xcb".

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, xcb.

Reinstalling the application may fix this problem.

**Comment by ogoffart** [comment]
@koroki: how about this one (from my attempt) https://hydro.woboq.im/owncloud/index.php/s/Dg3K2xmOhscZN8I

**Comment by koroki** [comment]
@ogoffart Other errors appers:
./ownCloud_desktop_sync_client_-x86_64.AppImage: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by ./ownCloud_desktop_sync_client_-x86_64.AppImage)
./ownCloud_desktop_sync_client_-x86_64.AppImage: /usr/lib64/libgpg-error.so.0: no version information available (required by /tmp/.mount_ownClodgzadn/usr/bin/../lib/libgcrypt.so.20)

**Comment by dragotin** [comment]
@ogoffart I was able to enable the nautilus plugins using a script as the AppRun script within the AppImage along this lines:

```bash
#!/bin/sh

echo "Start ownCloud Client"

EXT_PATH=$HOME/.local/share/nautilus-python/extensions

EXT_FILE=syncstate-ownCloud.py
HERE="$(dirname "$(readlink -f "${0}")")"

echo "Starting from $HERE"

if [ -L $EXT_PATH/$EXT_FILE ]; then
  rm $EXT_PATH/$EXT_FILE
fi

if [ ! -e $EXT_PATH/$EXT_FILE ]; then
  echo "=> Creating nautilus symlink"

  [ -d $EXT_PATH ] || mkdir -p $EXT_PATH
  ln -s ${HERE}/usr/share/nautilus-python/extensions/${EXT_FILE} ${EXT_PATH}/

  [ -d $HOME/.icons ] || mkdir $HOME/.icons
  ln -s ${HERE}/usr/share/icons/hicolor/256x256/apps/ownCloud*png $HOME/.icons
fi

export LD_LIBRARY_PATH=${HERE}/usr/lib:${HERE}/usr/lib64/:${HERE}/lib64:\
${HERE}/usr/lib/i386-linux-gnu/:\
${HERE}/usr/lib/x86_64-linux-gnu/:\
${HERE}/usr/lib32/:\
${HERE}/lib/i386-linux-gnu/:\
${HERE}/lib/x86_64-linux-gnu/

export XDG_DATA_DIRS=${HERE}/usr/share/

export QT_PLUGIN_PATH=${HERE}/usr/lib/qt5/plugins/:\
${HERE}/usr/lib/i386-linux-gnu/qt5/plugins/:\
${HERE}/usr/lib/x86_64-linux-gnu/qt5/plugins/:\
${HERE}/usr/lib32/qt5/plugins/:\
${HERE}/usr/lib64/qt5/plugins/

exec "${HERE}/usr/bin/owncloud" "$@"

```

Obviously, packages to enable the python extensions have to be available on the installed system. This AppRun script can certainly be improved to handle that as well, ie. proposing the user on first start of the AppImage to install the required packages.

**Comment by Lvaskz** [comment]
any progress?

**Comment by TheOneRing** [comment]
Fixed in 2.9 https://download.owncloud.com/desktop/ownCloud/daily/2.9/linux-appimage/?sort=time&order=desc
